addons/common/functions/fnc_addExplosionEventHandler.sqf

```diff
@@ -0,0 +1,38 @@
+#include "..\script_component.hpp"
+/*
+ * Author: johnb432, PabstMirror
+ * Adds event hander for explosions (for local projectiles)
+ * Warning: For shots that are tracers and explosive the event will trigger on all machines in range
+ *          This function may be changed after 2.18!
+ *
+ * Arguments:
+ * 0: Code to execute <CODE>
+ *
+ * Return Value:
+ * The CBA Event Handler Index <NUMBER>
+ *
+ * Example:
+ * [{systemChat "boom"}] call ace_common_fnc_addExplosionEventHandler
+ *
+ * Public: No (maybe after 2.18)
+ */
+
+params [["_code", {}, [{}]]];
+
+if !(missionNamespace getVariable [QGVAR(addExplosionEventHandlerActive), false]) then {
+    GVAR(addExplosionEventHandlerActive) = true;
+
```

Comment:
This might solve the issue of not adding the EH to existing ammo.

I haven't tested this, I've only tested `8 allObjects 2` which returns both mines and projectiles, but idk what it doesn't include and what else it include that we wouldn't want.

```suggestion

    // Add EH to existing projectiles and mines
    {
        // Rockets only explode on local clients
        _projectile addEventHandler ["Explode", {
            TRACE_1("exp",_this);
            [QGVAR(explosion), _this] call CBA_fnc_localEvent;
        }];
    } forEach ((8 allObjects 2) select {local _x});
```